pkg/controller/backup_controller.go

```diff
@@ -588,25 +597,47 @@ func (c *backupController) runBackup(backup *pkgbackup.Request) error {
 
 	var fatalErrs []error
 
-	// discovery cluster name or cluster kubeconfig name/namespace
-	// kubeconfig
-	// create clients from kubeconfig
-	// create discoveryHelper from kubeconfig
-
-	//backupper, err := backup.NewKubernetesBackupper(
-	//	s.veleroClient.VeleroV1(),
-	//	discoveryHelperTarget,
-	//	client.NewDynamicFactory(s.dynamicClientTarget),
-	//	podexec.NewPodCommandExecutor(s.kubeClientConfigTarget, s.kubeClientTarget.CoreV1().RESTClient()),
-	//	s.resticManager,
-	//	s.config.podVolumeOperationTimeout,
-	//	s.config.defaultVolumesToRestic,
-	//)
-	//if err := backupper.Backup(backupLog, backup, backupFile, actions, pluginManager); err != nil {
-	//	fatalErrs = append(fatalErrs, err)
-	//}
-
-	if err := c.backupper.Backup(backupLog, backup, backupFile, actions, pluginManager); err != nil {
+	clusterName := backup.StorageLocation.Name
+	kubeConfigSecret := &v1.Secret{}
```

Comment:
Just fyi. Cluster-api has a pretty nifty cluster cache tracker: https://github.com/kubernetes-sigs/cluster-api/blob/master/main.go#L219

I think it doesn't entirely fit our use case, but cluster-api has a very nifty: ClusterCacheTracker

Scope of ClusterCacheTracker is:
* keep a cache of kube clients for ClusterAPI clusters (they are created from secrets)
* do regular healthchecks

I think for our final implementation of Velero it's at least very useful to implement some of the patterns they are using there.

Reply:
Thanks. The current changes already includes a customized version of the tracker. Will force-push the changes to the branch soon :slightly_smiling_face: